src/engine/SCons/Tool/MSCommon/vc.py

```diff
@@ -705,17 +764,27 @@ def msvc_find_valid_batch_script(env, version):
     host_platform, target_platform, req_target_platform = platforms
     try_target_archs = [target_platform]
 
-    # VS2012 has a "cross compile" environment to build 64 bit
+    # Express versions have a "cross compile" environment to build 64 bit
     # with x86_amd64 as the argument to the batch setup script
+    # For 2015/2017 Express, this could extend to arm and arm64 targets.
     if req_target_platform in ('amd64', 'x86_64'):
         try_target_archs.append('x86_amd64')
-    elif not req_target_platform and target_platform in ['amd64', 'x86_64']:
-        # There may not be "native" amd64, but maybe "cross" x86_amd64 tools
-        try_target_archs.append('x86_amd64')
-        # If the user hasn't specifically requested a TARGET_ARCH, and
-        # The TARGET_ARCH is amd64 then also try 32 bits if there are no viable
-        # 64 bit tools installed
-        try_target_archs.append('x86')
+    elif req_target_platform in ('arm',):
+        try_target_archs.append('x86_arm')
+    elif req_target_platform in ('arm64',):
+        try_target_archs.append('x86_arm64')
+    elif not req_target_platform:
+        if target_platform in ('amd64', 'x86_64'):
+            # There may not be "native" amd64, but maybe cross x86_amd64 tools
+            try_target_archs.append('x86_amd64')
+            # If the user hasn't specifically requested a TARGET_ARCH,
+            # and the TARGET_ARCH is amd64 then also try 32 bits
+            # if there are no viable 64 bit tools installed
+            try_target_archs.append('x86')
+        elif target_platform in ('arm',):
```

Comment:
Can this happen? so target_platform not explicitly specified and host is x86 but target is arm?

Reply:
I suspect that's a good point. I wasn't paying attention to the requsted part.